### PR TITLE
feat: session management in settings

### PR DESCRIPTION
**Pull request by stefro**
## Summary
https://app.clickup.com/t/1px10qr

This is based on the original Fortify component. Style has been adapted to ark styling and tests have been added. The original component did not have any thorough testing.

## How to test
Go to `/account/settings/password`. It's the bottom section.

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

The lack of tests in the original component might be because of the way it reads the `browser session id`:
``` php
request()->session()->getId();
```
Due to the nature Livewire components are tested, there is no actual way of attaching a request to the component test.
Mocking a request is Laravel bad practice (according to the Laravel docs) and even if I would, this could get complicated. So I figured that `request()->session()->getId()` could easily be replaced by `session()->getId()` in this case, and this makes it possible to add tests.

I have found no valid reason to only read the browser ID from the request. But since this is a security feature, I would like other devs opinion on this.

### CODE REVIEW

**Comment by alfonsobries** [comment]
@stefro how can I manually test this?

Code looks good 👍

**Review by alfonsobries — APPROVED** [review]

**Comment by stefro** [comment]
@alfonsobries Good point! I've added the `how/where to test` to the summary now.

File: resources/images/modal/secure.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...6 12.219a2.459 2.459 0 001.62 0c29.611-10.316 29.147-34.76 28.808-52.607-.036-1.918-.071-3.73-.071-5.482a2.48 2.48 0 00-2.473-2.486c-10.58 0-18.635-3.056-25.349-9.618a2.464 2.464 0 00-3.449 0c-6.714 6.562-14.768 9.618-25.346 9.618a2.48 2.48 0 00-2.474 2.486c0 1.753-.035.915-.071 2.833"/><path stroke="#DBDEE5" stroke-width="2" d="M132.773 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603zm24 0c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603z"/><path fill="#E5F0F8" stroke="#007DFF" stroke-width="2" d="M108.394 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.604c2.771 0 5.018 2.233 5.018 4.988V68.44c0 2.755-2.247 4.988-5.018 4.988h-8.604z"/><path stroke="#007DFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M116.939 58.285l-5.369 7.612a.754.754 0 01-.63.353.864.864 0 01-.64-.31l-1.362-1.814"/></g><defs><clipPath id="a"><path fill="#fff" d="M0 0h120v94H0z" transform="translate(90.324 9.166)"/></clipPath></defs></svg>
```
Comment (leMaur) [inline resources/images/modal/secure.svg:1]: @stefro I suggest this version without ID `<clipPath id="a">`. Already optimized with svgomg.
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 302 112"><path fill="#DBDEE5" fill-rule="evenodd" d="M258.5 111.498v-14h-38v14h-2v-14h-21v-2h21v-11h2v11h38v-38h-36v-2h36v-38h-38v10h-2v-10h-31v-2h31v-15h2v15h38v-15h2v15h41v2h-41v38h21v2h-21v38h11v2h-11v14h-2zm-180 0v-14h-38v14h-2v-14h-8v-2h8v-37h-18v-2h18v-39H.5v-2h38v-15h2v15h38v-15h2v15h30v2h-30v10h-2v-10h-38v39h35v2h-35v37h38v-11h2v11h20v2h-20v14h-2zm207-10v-4h-4v-2h4v-4h2v4h4.001v2H287.5v4h-2zm-207-23v-4h-4v-2h4v-4h2v4h4v2h-4v4h-2zm-74-16v-4h-4v-2h4v-4h2v4h4v2h-4v4h-2zm214-15v-4h-4v-2h4v-4h2v4h4v2h-4v4h-2z" clip-rule="evenodd"/><path fill="#DBDEE5" d="M120.213 83.841a1 1 0 0 0 0-2v2zm58.471-40.859a1 1 0 0 0 0-2v2zm21.485-2a1 1 0 0 0 0 2v-2zm-7.895 40.86a1 1 0 0 0 0 2v-2zm-72.061 0H97.795v2h22.418v-2zm-22.419 0a4.837 4.837 0 0 1-4.837-4.838h-2a6.837 6.837 0 0 0 6.837 6.837v-2zm-4.837-4.838V47.82h-2v29.185h2zm0-29.185a4.837 4.837 0 0 1 4.837-4.837v-2a6.837 6.837 0 0 0-6.837 6.837h2zm4.837-5.837v1h80.891v-2H97.794v1zm102.375 1h3.274v-2h-3.274v2zm3.274 0a4.837 4.837 0 0 1 4.837 4.837h2a6.837 6.837 0 0 0-6.837-6.837v2zm4.837 4.837v29.185h2V47.82h-2zm0 29.185a4.837 4.837 0 0 1-4.837 4.837v2a6.837 6.837 0 0 0 6.837-6.837h-2zm-4.837 5.837v-1h-11.169v2h11.169v-1z"/><path stroke="#212225" stroke-linecap="round" stroke-width="2" d="M123.665 79.048c5.23 9.004 14.26 17.067 29.353 22.296a3.175 3.175 0 0 0 2.088 0c38.163-13.221 37.565-44.551 37.128-67.427-.047-2.458-.092-4.78-.092-7.027a3.187 3.187 0 0 0-3.188-3.186c-13.635 0-24.016-3.917-32.669-12.327a3.188 3.188 0 0 0-4.445 0c-8.652 8.41-19.032 12.327-32.666 12.327a3.187 3.187 0 0 0-3.188 3.186c0 2.247-.044 5.738-.091 8.196"/><path stroke="#007DFF" stroke-linecap="round" stroke-width="2" d="M133.51 80.372c4.006 4.736 10.191 9.117 19.096 12.219a2.459 2.459 0 0 0 1.62 0c29.611-10.316 29.147-34.76 28.808-52.607-.036-1.918-.071-3.73-.071-5.482a2.48 2.48 0 0 0-2.473-2.486c-10.58 0-18.635-3.056-25.349-9.618a2.464 2.464 0 0 0-3.449 0c-6.714 6.562-14.768 9.618-25.346 9.618a2.48 2.48 0 0 0-2.474 2.486c0 1.753-.035.915-.071 2.833"/><path stroke="#DBDEE5" stroke-width="2" d="M132.773 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603zm24 0c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603z"/><path fill="#E5F0F8" stroke="#007DFF" stroke-width="2" d="M108.394 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.604c2.771 0 5.018 2.233 5.018 4.988V68.44c0 2.755-2.247 4.988-5.018 4.988h-8.604z"/><path stroke="#007DFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="m116.939 58.285-5.369 7.612a.754.754 0 0 1-.63.353.864.864 0 0 1-.64-.31l-1.362-1.814"/></svg>
```

Comment (stefro) [reply resources/images/modal/secure.svg:1]: Thanks @leMaur for pointing that out! Today I've read the guidelines again and indeed, I was warned for this in https://ark.dev/docs/program-incentives/guidelines/frontend-frameworks#content-svg. I'll try to remember next time!

File: tests/database/migrations/2020_10_02_043202_create_sessions_table.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+final class CreateSessionsTable extends Migration
+{
+    public function up()
+    {
+        Schema::create('sessions', function (Blueprint $table) {
```
Comment (ItsANameToo) [inline tests/database/migrations/2020_10_02_043202_create_sessions_table.php:13]: doesn't this come with laravel by default? https://laravel.com/docs/8.x/session#database

File: resources/views/profile/logout-other-browser-sessions-form.blade.php [hunk 3]
```diff
@@ -1,99 +1,182 @@
...
-    <x-slot name="description">
-        @lang('ui::forms.logout-sessions.description')
-    </x-slot>
-
-    <x-slot name="content">
-        <div class="max-w-xl text-sm text-gray-600">
+<div dusk="logout-other-browser-sessions">
+    <div class="flex flex-col">
+        <span class="header-4">@lang('ui::forms.logout-sessions.title')</span>
+        <span class="mt-4">@lang('ui::forms.logout-sessions.description')</span>
```
Comment (ItsANameToo) [inline resources/views/profile/logout-other-browser-sessions-form.blade.php:4]: the other settings components use `ui::pages.user-settings.<section>` , better to keep the translations all in one place

File: resources/images/modal/secure.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...8.196"/><path stroke="#007DFF" stroke-linecap="round" stroke-width="2" d="M133.51 80.372c4.006 4.736 10.191 9.117 19.096 12.219a2.459 2.459 0 001.62 0c29.611-10.316 29.147-34.76 28.808-52.607-.036-1.918-.071-3.73-.071-5.482a2.48 2.48 0 00-2.473-2.486c-10.58 0-18.635-3.056-25.349-9.618a2.464 2.464 0 00-3.449 0c-6.714 6.562-14.768 9.618-25.346 9.618a2.48 2.48 0 00-2.474 2.486c0 1.753-.035.915-.071 2.833"/><path stroke="#DBDEE5" stroke-width="2" d="M132.773 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603zm24 0c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.603c2.772 0 5.019 2.233 5.019 4.988V68.44c0 2.755-2.247 4.988-5.019 4.988h-8.603z"/><path fill="#E5F0F8" stroke="#007DFF" stroke-width="2" d="M108.394 73.428c-2.772 0-5.019-2.233-5.019-4.988V55.613c0-2.755 2.247-4.988 5.019-4.988h8.604c2.771 0 5.018 2.233 5.018 4.988V68.44c0 2.755-2.247 4.988-5.018 4.988h-8.604z"/><path stroke="#007DFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M116.939 58.285l-5.369 7.612a.754.754 0 01-.63.353.864.864 0 01-.64-.31l-1.362-1.814"/></svg>
```
Comment (ItsANameToo) [inline resources/images/modal/secure.svg:1]: `fill` issue
image

File: resources/lang/en/forms.php [hunk 5]
```diff
@@ -72,6 +72,10 @@
         'description'    => 'Manage and logout your active sessions on other browsers and devices.',
         'content'        => 'If necessary, you may logout of all of your other browser sessions across all of your devices. If you feel your account has been compromised, you should also update your password',
         'confirm_logout' => 'Logout Other Browser Sessions',
+        'ip'             => 'IP',
+        'os'             => 'OS',
+        'browser'        => 'Browser',
+        'last_active'    => 'Last active',
```
Comment (ItsANameToo) [inline resources/lang/en/forms.php:78]:
```suggestion
        'last_active'    => 'Last Active',
```

File: tests/database/migrations/2020_10_02_043202_create_sessions_table.php (hunk 2 above)
Comment (stefro) [reply tests/database/migrations/2020_10_02_043202_create_sessions_table.php:13]: I thought so too, but it's not (anymore): https://github.com/laravel/laravel/tree/8.x/database/migrations

I needed this migration for tests.

File: resources/lang/en/forms.php [hunk 6]
```diff
@@ -72,6 +72,10 @@
         'description'    => 'Manage and logout your active sessions on other browsers and devices.',
```
Comment (ItsANameToo) [inline resources/lang/en/forms.php:72]: maybe we should add that it only shows the 3 most recent entries 🤔 will discuss with Sam

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: resources/lang/en/forms.php (hunk 6 above)
Comment (ItsANameToo) [reply resources/lang/en/forms.php:72]: Change the description to `Manage and logout your active sessions on other browsers and devices. Your most recent sessions are shown below.`

File: resources/views/profile/logout-other-browser-sessions-form.blade.php [hunk 7]
```diff
@@ -1,99 +1,182 @@
...
-                <!-- Other Browser Sessions -->
-                @foreach ($this->sessions as $session)
-                    <div class="flex items-center">
-                        <div>
-                            @if ($session->agent->isDesktop())
-                                <svg fill="none" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" viewBox="0 0 24 24" stroke="currentColor" class="w-8 h-8 text-gray-500">
-                                    <path d="M9.75 17L9 20l-1 1h8l-1-1-.75-3M3 13h18M5 17h14a2 2 0 002-2V5a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"></path>
-                                </svg>
+    @if (count($this->sessions) > 0)
+        <table class="hidden mt-8 w-full text-left table-auto md:table">
```
Comment (ItsANameToo) [inline resources/views/profile/logout-other-browser-sessions-form.blade.php:11]: the foundation library has table styling and components, can you use those instead?

File: resources/views/profile/logout-other-browser-sessions-form.blade.php (hunk 3 above)
Comment (stefro) [reply resources/views/profile/logout-other-browser-sessions-form.blade.php]: Done

File: resources/images/modal/secure.svg (hunk 4 above)
Comment (stefro) [reply resources/images/modal/secure.svg:1]: Solved

File: resources/lang/en/forms.php (hunk 5 above)
Comment (stefro) [reply resources/lang/en/forms.php]: Changed

File: resources/lang/en/forms.php (hunk 6 above)
Comment (stefro) [reply resources/lang/en/forms.php]: Changed

File: resources/views/profile/logout-other-browser-sessions-form.blade.php (hunk 7 above)
Comment (stefro) [reply resources/views/profile/logout-other-browser-sessions-form.blade.php]: Done

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**